lib/services/server_test.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+func TestNewAWSNodeFromEC2Instance(t *testing.T) {
+	isBadParameterErr := func(tt require.TestingT, err error, i ...interface{}) {
+		require.True(tt, trace.IsBadParameter(err), "expected bad parameter, got %v", err)
+	}
+
+	makeEC2Instance := func(fn func(*ec2Types.Instance)) ec2Types.Instance {
+		s := ec2Types.Instance{
+			PrivateDnsName:   aws.String("my-private-dns.compute.aws"),
+			InstanceId:       aws.String("i-123456789abcedf"),
+			VpcId:            aws.String("vpc-abcd"),
+			PrivateIpAddress: aws.String("172.31.1.1"),
+			Tags: []ec2Types.Tag{
```

Comment:
Are the labels that we use for these instances (account-id, region) somehow important from the security standpoint? If so, then it's crucial that the order of applying tags vs Teleport labels is maintained, and there should be a test that contains conflicting names on these.

Reply:
Yes, for sure.
Thank you :+1: